hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go

```diff
@@ -1551,6 +1551,58 @@ func TestPauseHostedControlPlane(t *testing.T) {
 	}
 }
 
+func TestHostedControlPlaneNoSecret(t *testing.T) {
+	fakeNonExistantSecret := "non-existant-secret"
+	fakeExistantSecret := "cluster-fake-pull-secret"
+	fakeHCPName := "cluster-fake"
+	fakeHCPNamespace := "master-cluster-fake"
+	testsCases := []struct {
+		name               string
+		hostedControlPlane *hyperv1.HostedControlPlane
+	}{
+		{
+			name: "if a HostedControlPlane references an existant pull secret it should progress correctly.",
+			hostedControlPlane: &hyperv1.HostedControlPlane{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: fakeHCPNamespace,
+					Name:      fakeHCPName,
+				},
+				Spec: hyperv1.HostedControlPlaneSpec{
+					PullSecret: corev1.LocalObjectReference{Name: fakeExistantSecret},
+				},
+			},
+		},
+		{
+			name: "if a HostedControlPlane references a non existant pull secret it should fail.",
+			hostedControlPlane: &hyperv1.HostedControlPlane{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: fakeHCPNamespace,
+					Name:      fakeHCPName,
+				},
+				Spec: hyperv1.HostedControlPlaneSpec{
+					PullSecret: corev1.LocalObjectReference{Name: fakeNonExistantSecret},
+				},
+			},
+		},
+	}
+	for _, tc := range testsCases {
```

Comment:
As far as I can see this not unit testing any of our logic. It's just storing and fetching an object within a fake client.

this func should be called `TestValidateReleaseImage`

Then within `t.Run(tc.name, func(t *testing.T) {` you need to call `ValidateReleaseImage`

Then for the fakeNonExistantSecret you expect it to return error, for fakeExistantSecret you expect it to not return error

For that you need a new field in your testCases struct i.e. expectedError.

For fakeExistantSecret you need to inject it into the fake client via WithObjects(

Reply:
Then just FYI if you wanted to go further you could wrap the logic which sets the reason into a function and unit test that as well. But let's stay with the above for now.

Reply:
Ohh there is already a test covering the non-existent PullSecret, let me delete the mine one and modify the already existent test case to check the semantics added.